GitHubTeams.ps1

```diff
@@ -276,6 +314,407 @@ filter Get-GitHubTeamMember
     return (Invoke-GHRestMethodMultipleResult @params | Add-GitHubUserAdditionalProperties)
 }
 
+function New-GitHubTeam
+{
+<#
+    .SYNOPSIS
+        Creates a team within an organization on GitHub.
+
+    .DESCRIPTION
+        Creates a team or within an organization on GitHub.
+
+        The Git repo for this module can be found here: http://aka.ms/PowerShellForGitHub
+
+    .PARAMETER OrganizationName
+        The name of the organization to create the team in.
+
+    .PARAMETER TeamName
+        The name of the team.
+
+    .PARAMETER Description
+        The description for the team.
+
+    .PARAMETER Maintainers
+        A list of GitHub IDs for organization members who will become team maintainers.
+
+    .PARAMETER RepositoryName
```

Comment:
We don't want users to be "surprised" by unexpected differences in behavior.  Throughout the rest of the module `RepositoryName` refers specifically to just the individual name of the repo, _not_ its full name which is what this parameter is _actually_ expecting.  I would suggest renaming this to something like `RepositoryFullName` and then updating the description of this comment to be a lot more clear on what is being expected.

Given this new property needed for API interaction, you may also want to update `Add-GitHubRepositoryAdditionalProperties` to also set a `RepositoryFullName` property that this could be then retrieve via the pipeline by property name.  If you do so, leave this as-is as a function, but then change to using the `begin/process/end` format to collect all of the repository full names so that you can use them in a single execution of the API.

Reply:
In fact, you probably want to do that anyway so that you could also support the pipelining in of multiple user objects if the user is trying to specify the maintainers via the pipeline instead.

Reply:
Regarding changes to the  `GitHubRepository` module being out-of-scope for this PR...

I think it's fine if you want to operate that way.  From my perspective, you're introducing an API that uses a new property from an existing object, so exposing that property for easy pipelining seems in-scope.  If you don't want to tackle that as part of this change, that's fine, but then I think we should get an issue opened to track having someone do that work as a follow-up.